pageserver/src/tenant/storage_layer/image_layer.rs

```diff
@@ -689,6 +689,22 @@ impl ImageLayerInner {
             };
         }
     }
+
+    #[allow(dead_code)]
+    pub(crate) fn iter<'a, 'ctx>(
+        &'a self,
+        ctx: &'ctx RequestContext,
+    ) -> ImageLayerIterator<'a, 'ctx> {
+        ImageLayerIterator {
+            image_layer: self,
+            ctx,
+            next_batch_start_key: Key::MIN,
+            key_values_batch: Vec::new(),
+            next_idx_in_batch: 0,
+            is_end: false,
+            batch_size: 1024, // The default value. Unit tests might use a different value. 1024 * 8K = 8MB buffer.
```

Comment:
image values can have any size, so i think it's better to use a batch size in terms of bytes instead of number of keys. `VectoredReadPlanner` also uses bytes for its `max_read_size` field.

Reply:
Fixed, b/c we want exactly one I/O, this is now enforced within the iterator.